test/integration/model/create.test.js

```diff
@@ -70,6 +70,28 @@ describe(Support.getTestDialectTeaser('Model'), function() {
       });
     }
 
+    it('should lock table during find and create', function () {
+      var User = this.sequelize.define('User', {
+        email: {
+          type: DataTypes.STRING
```

Comment:
Adding a unique: true here will fix the isuse.

I realize this is in no way clear from the documentation, but findOrCreate is only meant to work when we can detect duplicated records through a unique contraint violation